examples/vector_iteration.rs

```diff
@@ -1,14 +1,28 @@
+#[cfg(feature = "serde")]
+extern crate serde;
+
+#[cfg(not(feature = "serde"))]
 extern crate rustc_serialize;
 extern crate mustache;
 
 use std::str;
 use std::collections::HashMap;
 
-#[derive(RustcEncodable)]
+#[cfg_attr(not(feature = "serde"), derive(RustcEncodable))]
 struct User {
     name: String
 }
 
+#[cfg(feature = "serde")]
+impl serde::ser::Serialize for User {
```

Comment:
I think you deleted the other comment or else my github is broken, but I wrote a little bit about the hashmap behavior (or why not bother with `with_capacity(1)`). It can be summarised with a [short gist](https://play.rust-lang.org/?gist=9d7de03200c32313d75fa702bd57a871&version=stable&backtrace=0), I found it fun to look into the guts to ensure my trust in rust's defaults were sane!